src/modules/CRUD/components/Nav/Nav.js

```diff
@@ -54,32 +41,43 @@ export const Nav = ({ getAllLayersAction, resizingMap, layersList, t }) => {
           minimal
         />
       </div>
-      <ul
-        id="CRUD-nav__menu"
-        className="CRUD-nav__menu"
-      >
-        {layersListByGroups.map(({ group, list }) => (
-          <li className="CRUD-nav__group" key={group}>
-            <p className="CRUD-nav__group-name">
-              {group}
-            </p>
-            <ul className="CRUD-nav__list">
-              {list.map(({ name }) => (
-                <li className="CRUD-nav__item" key={name}>
-                  <NavLink
-                    to={generateURI('layer', { layer: name })}
-                  >
-                    <span className="bp3-button bp3-minimal">
-                      <Icon icon="layout" />
-                      <span className="bp3-button-text">{name}</span>
-                    </span>
-                  </NavLink>
-                </li>
-              ))}
-            </ul>
-          </li>
-        ))}
-      </ul>
+      { menu && (
+        <ul
+          id="CRUD-nav__menu"
+          className="CRUD-nav__menu"
+        >
+          {menu.sort(sortByOrder).map(({ name: group, pictogram: groupPictogram, crud_views: list }) => (
+            <li className="CRUD-nav__group" key={group}>
+              <p className="CRUD-nav__group-name">
+                {groupPictogram && (
```

Comment:
This block is used twice. Make it a component, and the rendering condition should goes inside it.

```jsx
const CrudNavIcon = ({ src, alt = '', ...rest }) => src && (
  <span className="bp3-icon" {...rest}>
    <img className="CRUD-nav__icon" src={src} alt={alt} />
  </span>
);
```

Reply:
Good point, done